controllers/rbac/aggregation.go

```diff
@@ -0,0 +1,262 @@
+	// oldRole.AggregationRule = newRole.AggregationRule
+	return a.Update(context.Background(), oldRole)
+}
+
+// func getRole(reader client.Reader, namespace string) *rbacv1.ClusterRole {
+func getRole() *rbacv1.Role {
+	fmt.Println("Got to getRole")
+	return &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleName,
+			Namespace: default_namespace,
+			// OwnerReferences: getOwnerRefs(reader, namespace),
+		},
+		// AggregationRule: &rbacv1.AggregationRule{
+		// 	ClusterRoleSelectors: []metav1.LabelSelector{
+		// 		{
+		// 			MatchLabels: map[string]string{
+		// 				aggregationLabelKey: "true",
+		// 			},
+		// 		},
+		// 	},
+		// },
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"*"}, // "get",
+				// "list",
+				// "watch",
+				// "create",
+				// "update",
+				// "patch",
+				// "delete",
+			},
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"*"}, // "get",
+				// "list",
+				// "watch",
+				// "create",
+				// "update",
+				// "patch",
+				// "delete",
+
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"events"},
+				Verbs: []string{
+					"create",
+					"patch",
+				},
+			},
+		},
+	}
+}
+
+func (a aggregation) createOrUpdateRoleBinding() error {
+	// check if the role exists
+	binding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
+			Namespace: default_namespace,
+		},
+	}
+	err := a.reader.Get(context.Background(), client.ObjectKeyFromObject(binding), binding)
+	if errors.IsNotFound(err) {
+		return a.createRoleBinding()
+	} else if err != nil {
+		return fmt.Errorf("failed to get  role binding: %v", err)
+	}
+	return a.updateRoleBinding(binding)
+}
+
+func (a aggregation) createRoleBinding() error {
+	err := a.Create(context.Background(), getRoleBinding(a.namespace))
+	return err
+}
+
+func (a aggregation) updateRoleBinding(oldBinding *rbacv1.RoleBinding) error {
+	newBinding := getRoleBinding(a.namespace)
+	oldBinding.RoleRef = newBinding.RoleRef
+	oldBinding.Subjects = newBinding.Subjects
+	return a.Update(context.Background(), oldBinding)
+}
+
+// func getRoleBinding(reader client.Reader, namespace string) *rbacv1.RoleBinding {
+func getRoleBinding(namespace string) *rbacv1.RoleBinding {
+	fmt.Println("Got to getRoleBinding")
+	return &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
+			Namespace: default_namespace,
+			// OwnerReferences: getOwnerRefs(reader, namespace),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     roleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      saName,
+				Namespace: namespace,
+			},
+		},
+	}
+}
+
+// func getOwnerRefs(reader client.Reader, namespace string) []metav1.OwnerReference {
```

Comment:
yeah
ownerreferences don't work cross namespace for namespaced resources
so, who does the cleanup?